frames/src/main/java/com/checkout/frames/component/cardholdername/CardHolderNameViewModel.kt

```diff
@@ -5,31 +5,63 @@ import androidx.compose.ui.text.input.ImeAction
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
 import com.checkout.base.mapper.Mapper
+import com.checkout.frames.R
 import com.checkout.frames.component.base.InputComponentState
 import com.checkout.frames.di.base.InjectionClient
 import com.checkout.frames.di.base.Injector
 import com.checkout.frames.di.component.CardHolderNameViewModelSubComponent
+import com.checkout.frames.screen.manager.PaymentStateManager
 import com.checkout.frames.style.component.CardHolderNameComponentStyle
 import com.checkout.frames.style.component.base.InputComponentStyle
 import com.checkout.frames.style.view.InputComponentViewStyle
+import kotlinx.coroutines.flow.update
 import javax.inject.Inject
 import javax.inject.Provider
 
 internal class CardHolderNameViewModel @Inject constructor(
+    private val paymentStateManager: PaymentStateManager,
     private val inputComponentStyleMapper: Mapper<InputComponentStyle, InputComponentViewStyle>,
     private val inputComponentStateMapper: Mapper<InputComponentStyle, InputComponentState>,
-    style: CardHolderNameComponentStyle,
+    style: CardHolderNameComponentStyle
 ) : ViewModel() {
 
     val componentState = provideViewState(style)
     val componentStyle = provideViewStyle(style.inputStyle)
+    val isInputFieldOptional = componentStyle.isInputFieldOptional
```

Comment:
can this be `private` or just use the value directly in [handleValidationResult]?

Reply:
Thanks for calling out. used directly value